### ISSUE TITLE
Improve transformation pipelines in 934xxx rules to catch post-base64Decode evasions

### ISSUE THREAD

**Issue by RedXanadu** [coreruleset/coreruleset#3311]
### Description

This problem was originally reported in #3182. It was fixed in #3203 _for the rule originally reported only_. Now we need to fix the remaining rules that are affected.

---

The transformation pipelines of the rules in REQUEST-934-APPLICATION-ATTACK-GENERIC.conf that make use of `base64Decode` need to be improved. We also need new tests to ensure that these rules detect the different evasion scenarios that they should: both *post* and *pre* Base64 decoding evasions should be caught.

The same series of anti-evasion transformations in these rules should be applied both **before** Base64 decoding takes place **and after** Base64 decoding has taken place.

This will fix the fact that we currently do not handle evasions that are present in a payload _after_ it has been Base64 decoded.

#### Further explanation:

- We currently _do_ detect:
  - Base64 encoded text that then has evasions applied to the characters of the _encoded_ text
- We currently _do not_ detect:
  - text that has evasions applied to it that is then Base64 encoded

> If you base64 the following payloads and send it as a cookie, only the first one will get blocked.
> ```
>{"rce":"_$$ND_FUNC$$_function(){...
> {"rce":"_$$\u004e\u0044_FUNC$$_\u0066unction(){...
> ```

---

Rules that need attention _(checked = rule fix is ready over in PR #3312)_:

- [x] 934100 _(this was the original issue, fixed in #3203)_
- [x] 934101
- [x] 934130
- [x] 934131
- [x] 934160
- [x] Measure before and after performance numbers, to make sure that expanding the transformation pipelines in conjunction with the `multiMatch` action isn't causing big issues

**Comment by RedXanadu** [comment]
Work is ongoing. I have data from 10,000 tests with and without the new transformations. I need to examine it.

**Comment by RedXanadu** [comment]
PR #3312 now ready for review and merge to resolve this issue.
